### PR TITLE
Fixes: P5-38 - Architecture Design (text)

### PR DESCRIPTION
**Pull request by Moesgaarda**
Den mangler stadig nogle småting der skal indsættes, men jeg tænker at det først bliver efter vi har fundet ud af hvor den egentlig hører hjemme henne.

### CODE REVIEW

File: report/3.Theory/Architecture.tex [hunk 1]
```diff
@@ -0,0 +1,29 @@
+\section{System Architecture}
+Before the implementation and building phases can begin, it is essential to give some thought towards the design of the system, both software and hardware wise.
+In order to do so, a component architecture diagram was constructed, which can be seen on Figure~\ref{fig:comparch}.
+%TODO Insert figure
+
+As seen on the diagram, the system will consist of two platforms: the LEGO NXT for handling the real time handling, and a Raspberry Pi for handling the machine intelligence.
```
Comment (nillas12) [inline report/3.Theory/Architecture.tex]: Handling the real time handling? Sounds weird

File: report/3.Theory/Architecture.tex [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+Each platform is split into two groups, a hardware group consisting of whatever hardware is connected to it and a software group which contains a series of classes that must be implemented for the device to function.
```
Comment (nillas12) [inline report/3.Theory/Architecture.tex]: Whatever hardware is connected to it? What is it?
Also I believe some commas are lacking here.

File: report/3.Theory/Architecture.tex [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+Both the hardware and software requirements are defined with a high level of abstraction, since the diagram should be accurate regardless of implementation details or choice of hardware to a large extend.
```
Comment (nillas12) [inline report/3.Theory/Architecture.tex]: I don't think "to a large extend" is required

File: report/3.Theory/Architecture.tex [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+\item Get data from camera
+\item Run image recognition algorithm
+\item Send result data to NXT
+\end{enumerate}
+
+Due to the dependencies of these methods, none of them can run independently without the result of the prior method, since it would simply lead to re-sending the same information twice.
+Furthermore, it is important to notice that the diagram contains methods for both shooting a laser, and a projectile as mentioned in the requirements, Section~\ref{sec:moscow}.
+%TODO lav et label ved moscow
+As the projectile firing model requires additional calculations, the device will function with merely the laser methods and hardware. 
+The methods required projectile firing are colored red to indicate this.
```
Comment (nillas12) [inline report/3.Theory/Architecture.tex]: The methods required? Don't you mean requiring?

File: report/3.Theory/Architecture.tex [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+%TODO fix colors
+With the architecture in place, it is now possible to foxus on the details of what is required for implementing each part of the architecture.
```
Comment (nillas12) [inline report/3.Theory/Architecture.tex]: foxus to focus

File: report/3.Theory/0.TheoryCtrl.tex [hunk 6]
```diff
@@ -2,3 +2,4 @@ \chapter{Theory}
 \input{3.Theory/1.TrackingObjects}
 \input{3.Theory/ObjectRecognition.tex}
 \input{3.Theory/2.RTS}
+\input{3.Theory/Architecture.tex}
```
Comment (C0DK) [inline report/3.Theory/0.TheoryCtrl.tex]: den skal da vist lige i documentation hva

Comment (Moesgaarda) [reply report/3.Theory/0.TheoryCtrl.tex]: det er da vist nok rigtigt hva

**Pull request merged**